src/compile_fail_tests.md

```diff
@@ -9,4 +9,27 @@ use ::nougat::*;
 trait Foo {}
 ```
 
+### The `#[gat]` attribute only allows one item in imports
+
+```rust ,compile_fail
+// for some reason we need to surround this in a module,
+// otherwise the inner module imports don't resolve correctly,
+// and the compile failure is not the right failure we test.
```

Comment:
Ah yeah, you'd need to add a dummy `fn main() {}`, otherwise doctest heuristics generate their own and wrap the whole code within the fn body braces, and at that point the items defined therein cannot be named by qualified paths:

```rust
fn main() {
    enum Thing {}
    mod example { use super::Thing; } // Error
}
```

But if your code contains a `fn main() {}` it won't do that IIRC